File: palindrome.js
function palindrome(str) {
    const lowerCaseStr = str.toLowerCase();
    const middle = Math.ceil(str.length / 2);

    for(let i = 0; i < middle; i++) {
        if (lowerCaseStr[i] !== lowerCaseStr[str.length - 1 - i]) {
            return false;
        }
    }

    return true;
}

console.clear();
console.log(palindrome('racecar'));
console.log(palindrome('raccar'));
console.log(palindrome('😀raccar'));
console.log(palindrome('12321'));

File: palindrome.py
#console.log(palindrome('racecar'))
#console.log(palindrome('😀raccar😀'))
#console.log(palindrome('😀raccar'));
#console.log(palindrome('12321'));
first_pal = 'racecar'
second_pal = ('😀raccar😀')
print(second_pal[1])

def is_palindrome(input_str):
    return input_str == input_str[::-1]

print(is_palindrome(first_pal))
print(is_palindrome(second_pal))
